CIP-?/README.md

```diff
@@ -0,0 +1,109 @@
+---
+CIP: ?
+Title: URI Scheme - Block and transaction objects
+Status: Proposed
+Category: Tools
+Authors:
+    - Pi Lanningham <pi@sundaeswap.finance>
+Implementors: N/A
+Discussions:
+    - https://github.com/cardano-foundation/CIPs/pull/556
+Created: 2020-09-22
+License: CC-BY-4.0
+---
+
+## Abstract
+
+This extends CIP-13, which describes a Cardano URI scheme, with several more objects that would be useful to be able to assign addresses to, in particular blocks, transactions, transaction metadata, and transaction outputs.
+
+## Motivation: why is this CIP necessary?
+
+CIP-13 defined two initial URL authorities, for payment links and delegating to a stakepool.
+
+However, in a number of contexts, it would be useful to canonically link to other Cardano objects, such as:
+- Providing links to a transaction, to be opened in a wallet or chain explorer of the users choice
+- To provide richly interconnected metadata, such as in [CIP-100](https://github.com/cardano-foundation/CIPs/pull/556)
+
+Without a canonical standard for how to define these URIs, these objects are either unaddressable, not machine-interpretable, or will suffer from a divergence of convention. Similarly, we seek to fit an existing structure, such as CIP-0013, to reduce the number of different conventions that need be supported by the ecosystem.
+
+## Specification
+
+We extend CIP-13 with 2 new authorities for referencing blocks and transactions.
+
+Examples:
+```
+web+cardano://block/c6a8976125193dfae11551c5e80a217403d953c08ebbd69bba904d990854011f
+web+cardano://transaction/7704a68404facf7126fa356f1b09f0e4c552aeef454cd0daba4208f3a64372e9
+web+cardano://transaction/7704a68404facf7126fa356f1b09f0e4c552aeef454cd0daba4208f3a64372e9#1
+web+cardano://transaction/7704a68404facf7126fa356f1b09f0e4c552aeef454cd0daba4208f3a64372e9/metadata
+web+cardano://transaction/d3616b772c91f118346e74863d1722810a4858e4d7cc7663dc2eed345d7bca72/metadata/674
+web+cardano://transaction/self/metadata/1694
+```
+
+### Grammar & interpretation
+
+We extend the grammar from CIP-0013 with two new authorities:
+
+```
+authorityref = (stakepoolref | blockref | transactionref | otherref)
+```
+
+#### Block queries
+
+A link referencing a block by either block hash or height.
+
+> WARNING: Referencing blocks by height is provided in the case of extremely tight space requirements, but referencing a recent block (within the rollback horizon of the chain) is unstable.  Due to chain re-orgs, it may refer to different content for different users, or at different times. This should only be used when this is not critical, or for blocks outside of the rollback horizon.
+
+```
+blockref = "//block" query
+query = ( "?block=" block_hash | "?height=" block_height)
+
+block_hash = 64HEXDIG
+block_height = *digit
+```
```

Comment:
I don't think we would ever run into "space" constraints with this URI schema so I would prefer to see that all blocks be referenced by their cryptographic hash rather than a specific numeric index that is fragile for the aforementioned reasons.

Reply:
You'd be surprised; Currently the `anchor` field in CIP-1694 is limited to 64 characters, for example.

Reply:
I would say leave it off, because I don't currently have a strong motivation for it; but if we do, the next step of simplification makes it hard to add later; `web+cardano://block/{block_hash}` doesn't have a clear way to represent height.

Reply:
Since this is rolling out in a governance context, wouldn't a human readable `height` be important to compare dates of governance propositions or decisions (even with lack of finality sometimes applicable)?  I can see how could be a "strong motivation" although @Quantumplation you would have a better idea of that from the workshops.

Reply:
Not really; you could still grab that data on-chain, or store a timestamp in the governance metadata, etc.

This is more so about generating a URL *only* based on the height, like being able to link to `web+cardano://block?height=9664256` vs `web+cardano://block?hash=4bfe21ff68e22266c14bcbdce6fc76d31c9813516c4bcdfa697069bb5e9e1f41` or `web+cardano://block/4bfe21ff68e22266c14bcbdce6fc76d31c9813516c4bcdfa697069bb5e9e1f41`

From a data availability standpoint they're the same, it's just a matter of what URLs people produce, read, and click on.

Reply:
@Crypto2099 what do you think, what's the verdict here?

Reply:
I am content to leave the `height=` syntax in given that we have provided a disclaimer that this cannot/should not be trusted within the rollback horizon of the chain.